<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>397</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="QWidget" name="widget" native="true">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>381</width>
     <height>531</height>
    </rect>
   </property>
   <widget class="QDialogButtonBox" name="buttonBox">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>450</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="standardButtons">
     <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
    </property>
   </widget>
   <widget class="QStackedWidget" name="model_operation">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>85</y>
      <width>371</width>
      <height>101</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">#model_operation {
border:1px solid rgba(0, 0, 0, 0.23);
}</string>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="page_import_training_model">
     <widget class="QLineEdit" name="lineEdit_sel_model">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>10</y>
        <width>271</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="placeholderText">
       <string>模型文件</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_sel_model">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>选择模型</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_sel_wellname">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>40</y>
        <width>241</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="placeholderText">
       <string>wellname</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_38">
      <property name="geometry">
       <rect>
        <x>16</x>
        <y>40</y>
        <width>60</width>
        <height>19</height>
       </rect>
      </property>
      <property name="text">
       <string>选择预测井</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_sel_well">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>39</y>
        <width>22</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>../image/import_right.png</normaloff>../image/import_right.png</iconset>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="page_custom_training_model">
     <widget class="QComboBox" name="comboBox_model">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>40</y>
        <width>271</width>
        <height>22</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>KNN</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>SVM</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Boosting类算法</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>随机森林</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>XGBoost</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>LSTMClassifier</string>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>16</x>
        <y>40</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>20</height>
       </size>
      </property>
      <property name="text">
       <string>算法选择：</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_import_lithofacies_data">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>10</y>
        <width>271</width>
        <height>21</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_import_lithofacies_data">
      <property name="geometry">
       <rect>
        <x>2</x>
        <y>10</y>
        <width>81</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>导入岩相数据</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_75">
      <property name="geometry">
       <rect>
        <x>16</x>
        <y>70</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>20</height>
       </size>
      </property>
      <property name="text">
       <string>模型名称</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_modelname">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>70</y>
        <width>271</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>DefaultModelName</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_param_set">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>220</y>
      <width>371</width>
      <height>231</height>
     </rect>
    </property>
    <property name="title">
     <string>模型参数设置</string>
    </property>
    <widget class="QStackedWidget" name="modeParamSet">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>371</width>
       <height>191</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">/* background-color:rgb(255, 170, 0);*/
#modeParamSet{
border:1px solid rgba(0, 0, 0, 0.23);
}
</string>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="page_KNN">
      <widget class="QWidget" name="formLayoutWidget_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>261</width>
         <height>171</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_2">
        <item row="0" column="0">
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>邻居数量</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="n_neighbors">
          <property name="value">
           <number>5</number>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_12">
          <property name="text">
           <string>weights</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QCheckBox" name="weights">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>distance</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_13">
          <property name="text">
           <string>algorithm</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QCheckBox" name="algorithm">
          <property name="text">
           <string>auto</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_35">
          <property name="text">
           <string>leaf_size</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QSpinBox" name="leaf_size">
          <property name="value">
           <number>30</number>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_14">
          <property name="text">
           <string>欧几里得距离</string>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="label_15">
          <property name="text">
           <string>距离度量</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QCheckBox" name="metric">
          <property name="text">
           <string>minkowski</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="QLabel" name="label_16">
          <property name="text">
           <string>K折交叉验证</string>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QSpinBox" name="cv_folds">
          <property name="value">
           <number>5</number>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QSpinBox" name="p">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>2</number>
          </property>
          <property name="value">
           <number>2</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QWidget" name="page_SVM">
      <widget class="QWidget" name="formLayoutWidget">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>181</width>
         <height>71</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>正则化参数C</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>gamma</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>K折交叉验证</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="SVM_C">
          <property name="value">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="SVM_gamma">
          <property name="value">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="SVM_cv_folds">
          <property name="value">
           <number>5</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QWidget" name="page_Boosting">
      <widget class="QWidget" name="formLayoutWidget_3">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>181</width>
         <height>71</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_3">
        <item row="0" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>最大迭代次数</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>学习率</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>学习器的最大深度</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="Boosting_max_depth">
          <property name="value">
           <number>3</number>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QDoubleSpinBox" name="Boosting_learning_rate">
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="value">
           <double>0.100000000000000</double>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="Boosting_n_estimators">
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>100</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QWidget" name="page_RandomForest">
      <widget class="QWidget" name="formLayoutWidget_4">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>181</width>
         <height>86</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_4">
        <item row="0" column="0">
         <widget class="QLabel" name="label_20">
          <property name="text">
           <string>随机森林中树的数量</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_21">
          <property name="text">
           <string>树的最大深度</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_22">
          <property name="text">
           <string>随机数生成器</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="RandomForest_n_estimators">
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>100</number>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="RandomForest_max_depth">
          <property name="value">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="RandomForest_random_state">
          <property name="value">
           <number>42</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QWidget" name="page_XGBoost">
      <widget class="QWidget" name="formLayoutWidget_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>219</width>
         <height>171</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_5">
        <item row="0" column="0">
         <widget class="QLabel" name="label_23">
          <property name="text">
           <string>学习率</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_24">
          <property name="text">
           <string>树的个数</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_25">
          <property name="text">
           <string>树的深度</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_26">
          <property name="text">
           <string>决定最小叶子节点样本权重和</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_27">
          <property name="text">
           <string>每棵随机采样的比例</string>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="label_36">
          <property name="text">
           <string>subsample</string>
          </property>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="QLabel" name="label_37">
          <property name="text">
           <string>colsample_bytree</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="XGBoost_n_estimators">
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>100</number>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="XGBoost_max_depth">
          <property name="value">
           <number>5</number>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QSpinBox" name="XGBoost_min_child_weight">
          <property name="autoFillBackground">
           <bool>true</bool>
          </property>
          <property name="value">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QSpinBox" name="XGBoost_gamma">
          <property name="value">
           <number>5</number>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDoubleSpinBox" name="doubleSpinBox">
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="value">
           <double>0.100000000000000</double>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QDoubleSpinBox" name="XGBoost_subsample">
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="value">
           <double>0.800000000000000</double>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QDoubleSpinBox" name="XGBoost_colsample_bytree">
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="value">
           <double>0.800000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QWidget" name="page_LSTMClassifier">
      <widget class="QWidget" name="formLayoutWidget_6">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>231</width>
         <height>175</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_6">
        <item row="0" column="0">
         <widget class="QLabel" name="label_28">
          <property name="text">
           <string>隐藏层大小</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_29">
          <property name="text">
           <string>层的数量</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="LSTMClassifier_num_layers">
          <property name="value">
           <number>2</number>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_30">
          <property name="text">
           <string>丢弃率</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QDoubleSpinBox" name="LSTMClassifier_dropout">
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="value">
           <double>0.200000000000000</double>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_31">
          <property name="text">
           <string>最大迭代次数</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QSpinBox" name="LSTMClassifier_epochs">
          <property name="maximum">
           <number>20000</number>
          </property>
          <property name="value">
           <number>1000</number>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_32">
          <property name="text">
           <string>学习率</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QDoubleSpinBox" name="LSTMClassifier_learning_rate">
          <property name="decimals">
           <number>4</number>
          </property>
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="value">
           <double>0.001000000000000</double>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QComboBox" name="LSTMClassifier_optimizer">
          <item>
           <property name="text">
            <string>Adam</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>SGD</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>RMSprop</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Adagrad</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Adamax</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="label_33">
          <property name="text">
           <string>损失函数</string>
          </property>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="QLabel" name="label_34">
          <property name="text">
           <string>优化器</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QComboBox" name="LSTMClassifier_loss_function">
          <item>
           <property name="text">
            <string>CrossEntropyLoss</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>MSELoss</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>BCELoss</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>NLLLoss</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>SmoothL1Loss</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="LSTMClassifier_hidden_dim">
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>128</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>117</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string>模型操作</string>
    </property>
    <widget class="QRadioButton" name="import_training_model">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>91</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>导入模型预测</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="custom_training_model">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>111</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>自定义训练模型</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Dialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Dialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
